### PR TITLE
[client] Avoid panic when there is no conn client

### PR DESCRIPTION
**Pull request by mlsmaycon**
### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: client/internal/connect.go [hunk 1]
```diff
@@ -297,6 +297,9 @@ func (c *ConnectClient) run(
 }
 
 func (c *ConnectClient) Engine() *Engine {
+	if c.engine == nil {
```
Comment (lixmal) [inline client/internal/connect.go]: This needs to be inside the lock. But I don't see the sense of this, if `c.engine` is nil then we already return nil.

File: client/internal/connect.go [hunk 2]
```diff
@@ -305,6 +308,9 @@ func (c *ConnectClient) Engine() *Engine {
 }
 
 func (c *ConnectClient) Stop() error {
+	if c.engine == nil {
```
Comment (lixmal) [inline client/internal/connect.go]: This needs to be inside the lock

Comment (mlsmaycon) [reply client/internal/connect.go]: checked the wrong obj. fixed now

File: client/internal/connect.go (hunk 1 above)
Comment (mlsmaycon) [reply client/internal/connect.go]: checked the wrong obj. now its fixed

**Pull request merged**